open_optical_gating/cli/determine_reference_period.py

```diff
@@ -83,13 +103,27 @@ def establish_indices(sequence, settings, padding=True):
 
 
 def calculate_period_length(diffs):
+    """ Attempt to determine the period of one heartbeat, from the diffs array provided. The period will be measured backwards from the most recent frame in the array
+        Parameters:
+            diffs    ndarray    Diffs between latest frame and previously-received frames
+        Returns:
+            Period, or -1 if no period found
+        """
+    
+    # TODO: JT writes: the comment below can probably be removed now I have written the above function header. BUT, can somebody confirm or deny
+    # the bit about "the list is in reverse order"? That does not seem consistent with e.g. "bestMatchEntry = diffs.size - bestMatchPeriod", "score = diffs[diffs.size - d]" etc.
+    # Note that I have added comments about "chronological order" to earlier functions, above, based on my code-reading (which matches my expectations),
+    # but I started to doubt myself here when I found an explicit comment saying "reverse order"...
+    
     # Calculate the heart period (with sub-frame interpolation) based on a provided list of comparisons between the current frame and previous frames. The list is in reverse order (i.e. difference with most recent frame comes first)
-    bestMatchPeriod = estimate_period_length(diffs)
+    bestMatchPeriod = estimate_integer_period_length(diffs)
     bestMatchEntry = diffs.size - bestMatchPeriod
 
     if bestMatchPeriod == -1:
         return -1
 
+    # TODO: JT writes: isn't it slightly weird that the v_fitting function lives in the POG module?
+    # Feel free just to delete this comment if you don't have any better ideas, though!
```

Comment:
This was a random choice. I had two, identical v_fitting functions - one in pog and one here. They were both used once in their respective modules. I decided having two was silly and deleted one. Turns out it was this one. Happy or it to be in either really.